lib/IRGen/IRGenSIL.cpp

```diff
@@ -1116,6 +1120,193 @@ class IRGenSILFunction :
+                                        SILBasicBlock &_entry,
+                                        Explosion &_allParamValues)
+      : SyncEntryPointArgumentEmission(_IGF, _entry, _allParamValues){};
+};
+
+class SyncNativeCCEntryPointArgumentEmission final
+    : public NativeCCEntryPointArgumentEmission,
+      public SyncEntryPointArgumentEmission {
+public:
+  SyncNativeCCEntryPointArgumentEmission(IRGenSILFunction &_IGF,
+                                         SILBasicBlock &_entry,
+                                         Explosion &_allParamValues)
+      : SyncEntryPointArgumentEmission(_IGF, _entry, _allParamValues){};
+
+  llvm::Value *getErrorResultArgument() override {
+    return allParamValues.takeLast();
+  }
+  llvm::Value *getContext() override { return allParamValues.takeLast(); }
+  Explosion getArgumentExplosion(unsigned index, unsigned size) override {
+    assert(size > 0);
+    Explosion result;
+    allParamValues.transferInto(result, size);
+    return result;
+  }
+  llvm::Value *getSelfWitnessTable() override {
+    return allParamValues.takeLast();
+  }
+  llvm::Value *getSelfMetadata() override { return allParamValues.takeLast(); }
+  llvm::Value *getCoroutineBuffer() override {
+    return allParamValues.claimNext();
+  }
+};
+
+class AsyncNativeCCEntryPointArgumentEmission final
+    : public NativeCCEntryPointArgumentEmission,
+      public AsyncEntryPointArgumentEmission {
+  llvm::Value *context;
+  /*const*/ AsyncContextLayout layout;
+  const Address dataAddr;
+
+public:
+  AsyncNativeCCEntryPointArgumentEmission(IRGenSILFunction &IGF,
+                                          SILBasicBlock &entry,
+                                          Explosion &allParamValues)
+      : AsyncEntryPointArgumentEmission(IGF, entry, allParamValues),
+        context(allParamValues.claimNext()), layout(getAsyncContextLayout(IGF)),
+        dataAddr(layout.emitCastTo(IGF, context)){};
+
+  llvm::Value *getErrorResultArgument() override {
+    auto errorLayout = layout.getErrorLayout();
+    Address addr = errorLayout.project(IGF, dataAddr, /*offsets*/ llvm::None);
+    // FIXME: FIXME_NOW: Alloca a %swift.error* and store the contents of
+    //                   addr to it.
+    return addr.getAddress();
+  }
+  llvm::Value *getContext() override {
+    auto contextLayout = layout.getLocalContextLayout();
+    Address addr = contextLayout.project(IGF, dataAddr, /*offsets*/ llvm::None);
+    auto &ti = cast<LoadableTypeInfo>(contextLayout.getType());
+    Explosion explosion;
+    // FIXME: FIXME_NOW: CORRECTNESS: Actually use these fields.
+    bool needsAllocas;
+    SmallVector<AddressToDeallocate, 4> addressesToDeallocate;
+    bool dependsOnContextLifetime;
+    loadArgumentFromContext(explosion, layout.getLocalContextType(), ti, addr,
+                            layout.getLocalContextConvention(), IGF,
+                            IGF.CurSILFn->getLoweredFunctionType(),
```

Comment:
I am confused by this line. Shouldn't that be `/*isNoEscape*true`?

Reply:
It definitely should be.  Fixed.